File: src/main/java/cn/bromine0x23/plantuml/controllers/render/EpsController.java
package cn.bromine0x23.plantuml.controllers.render;

import net.sourceforge.plantuml.FileFormat;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 渲染PlantUML为EPS格式
 *
 * @author <a href="mailto:bromine0x23@163.com">Bromine0x23</a>
 */
@RestController
@RequestMapping({"/eps"})
public class EpsController extends AbstractRenderController {

	private static final MediaType CONTENT_TYPE = MediaType.parseMediaType("application/postscript");

	@Override
	protected FileFormat getFileFormat() {
		return FileFormat.EPS;
	}

	@Override
	protected MediaType getContentType() {
		return CONTENT_TYPE;
	}
}

File: src/main/kotlin/cn/bromine0x23/plantuml/controllers/EpsController.kt
package cn.bromine0x23.plantuml.controllers

import cn.bromine0x23.plantuml.services.RenderService
import cn.bromine0x23.plantuml.services.UmlService
import net.sourceforge.plantuml.FileFormat
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/eps")
class EpsController(
	umlService: UmlService,
	renderService: RenderService
): AbstractRenderController(umlService, renderService) {

	companion object {
		@JvmStatic
		private val CONTENT_TYPE = MediaType.parseMediaType("application/postscript")
	}

	override val format: FileFormat
		get() = FileFormat.EPS

	override val contentType: MediaType
		get() = CONTENT_TYPE
}